tools/integration_tests/readonly/list_objects_test.go

```diff
@@ -35,20 +35,25 @@ func TestListObjectsInBucket(t *testing.T) {
 		log.Fatal(err)
 	}
 
+	// Comparing number of objects in the testBucket - 2
 	if len(obj) != NumberOfObjectsInTestBucket {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != DirectoryNameInTestBucket && obj[0].IsDir() != true {
+	// Comparing first object name and type
+	// Name - testBucket/Test, Type - Dir
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
 		t.Errorf("Listed object is incorrect.")
```

Comment:
nit: Object Listed for bucket directory is incorrect.

Reply:
fixed